app/services/foreman/plugin.rb

```diff
@@ -175,6 +175,14 @@ def delete_menu_item(menu, item)
       Menu::Manager.map(menu).delete(item)
     end
 
+    # Extends an already registered extensible page.
+    # Usage:
+    # extend_page "smart_proxies/show", :tabbed_menu, :name => "Example Tab", :partial => "path/to/partial", :priority => 10000
+    # extend_page "smart_proxies/show", :foreman_details, :name => "Example Content", :partial => "path/to/partial"
+    def extend_page(page_name, mountpoint, opts)
+      Pagelets::Manager.add_pagelet(page_name, mountpoint, opts)
+    end
```

Comment:
Nit pick - you could do `extend_page` DSL block for a nicer code, but not necessary really. This is fine.